InvenTree/InvenTree/tasks.py

```diff
@@ -51,22 +51,54 @@ def schedule_task(taskname, **kwargs):
         pass
 
 
-def offload_task(taskname, *args, **kwargs):
+def offload_task(taskname, force_sync=False, *args, **kwargs):
     """
-    Create an AsyncTask.
-    This is different to a 'scheduled' task,
-    in that it only runs once!
+        First check if the task method pointed
+        by taskname is implemented inside this file.
+
+        Then create an AsyncTask if workers are running.
+        This is different to a 'scheduled' task,
+        in that it only runs once!
+
+        If workers are not running or force_sync flag
+        is set then the task is ran synchronously.
     """
 
+    # Get task list
+    tasks = get_task_list()
+
+    # Check if task exists
+    if taskname not in tasks:
```

Comment:
I don't want to limit this to *only* tasks within this particular file. We may in the future have a lot of async tasks split between the various apps, e.g.

`stock.tasks.doSomeTask`

Reply:
Umm not sure how can we guarantee the task is implemented in this case, any idea?

Reply:
You should have a look at how django-q manages this

Reply:
@SchrodingersGat Please take a look at the new implementation, I'm sure it will be much more to your liking :smiley: